Tasks/AzureAppServiceManage/azureappservicemanage.ts

```diff
@@ -66,6 +68,16 @@ async function run() {
                 console.log(await azureRmUtil.stopAppService(endPoint, resourceGroupName, webAppName, specifySlotFlag, slotName));
                 break;
             }
+            case "Install Extensions": {
+                resourceGroupName = (specifySlotFlag ? resourceGroupName : await azureRmUtil.getResourceGroupName(endPoint, webAppName));
+                var publishingProfile = await azureRmUtil.getAzureRMWebAppPublishProfile(endPoint, webAppName, resourceGroupName, specifySlotFlag, slotName);
+                tl.debug('Retrieved publishing Profile');
+                var anyExtensionInstalled = await extensionManage.installExtensions(publishingProfile, extensionList.split(','));
+                if(!anyExtensionInstalled) {
+                    tl.debug('No new extension installed. Skipping Restart App Service.');
```

Comment:
We restart the app service, only when extensions are installed.
If extension is already installed, then we skip the installation and no restart done.